provisioning/plugins/corim-nitro-decoder/extractor.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2022 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+package main
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/veraison/corim/comid"
+	"github.com/veraison/services/proto"
+	structpb "google.golang.org/protobuf/types/known/structpb"
+)
+
+type Extractor struct{}
+
+func (o Extractor) SwCompExtractor(rv comid.ReferenceValue) ([]*proto.Endorsement, error) {
+	return nil, fmt.Errorf("Not implemented, not needed?")
+	// var nitroClassAttrs NitroClassAttributes
+
+	// if err := nitroClassAttrs.FromEnvironment(rv.Environment); err != nil {
+	// 	return nil, fmt.Errorf("could not extract PSA class attributes: %w", err)
+	// }
+
+	// // Each measurement is encoded in a measurement-map of a CoMID
+	// // reference-triple-record.  Since a measurement-map can encode one or more
+	// // measurements, a single reference-triple-record can carry as many
+	// // measurements as needed, provided they belong to the same PSA RoT
+	// // identified in the subject of the "reference value" triple.  A single
+	// // reference-triple-record SHALL completely describe the updatable PSA RoT.
+	// swComponents := make([]*proto.Endorsement, 0, len(rv.Measurements))
+
+	// for i, m := range rv.Measurements {
+	// 	var nitroSwCompAttrs NitroSwCompAttributes
+
+	// 	if err := nitroSwCompAttrs.FromMeasurement(m); err != nil {
+	// 		return nil, fmt.Errorf("extracting measurement at index %d: %w", i, err)
+	// 	}
+
+	// 	swAttrs, err := makeSwAttrs(nitroClassAttrs, nitroSwCompAttrs)
+	// 	if err != nil {
+	// 		return nil, fmt.Errorf("failed to create software component attributes: %w", err)
+	// 	}
+
+	// 	swComponent := proto.Endorsement{
+	// 		Scheme:     proto.AttestationFormat_AWS_NITRO,
+	// 		Type:       proto.EndorsementType_REFERENCE_VALUE,
+	// 		Attributes: swAttrs,
+	// 	}
+
+	// 	swComponents = append(swComponents, &swComponent)
+	// }
+
+	// if len(swComponents) == 0 {
+	// 	return nil, fmt.Errorf("no software components found")
+	// }
+
+	// return swComponents, nil
+}
+
+func makeSwAttrs(c NitroClassAttributes, s NitroSwCompAttributes) (*structpb.Struct, error) {
+	return nil, fmt.Errorf("Not implemented, not needed?")
+	// swAttrs := map[string]interface{}{
+	// 	//"psa.impl-id":           c.ImplID,
+	// 	"psa.signer-id":         s.SignerID,
+	// 	"psa.measurement-value": s.MeasurementValue,
+	// 	"psa.measurement-desc":  s.AlgID,
+	// }
+
+	// if c.Vendor != "" {
+	// 	swAttrs["psa.hw-vendor"] = c.Vendor
+	// }
+
+	// if c.Model != "" {
+	// 	swAttrs["psa.hw-model"] = c.Model
+	// }
+
+	// if s.MeasurementType != "" {
+	// 	swAttrs["psa.measurement-type"] = s.MeasurementType
+	// }
+
+	// if s.Version != "" {
+	// 	swAttrs["psa.version"] = s.Version
+	// }
+
+	// return structpb.NewStruct(swAttrs)
+}
+
+func (o Extractor) TaExtractor(avk comid.AttestVerifKey) (*proto.Endorsement, error) {
```

Comment:
this instead should not be needed: the trust anchor is fixed and can be embedded in the package that provides the verification logics

Reply:
The trust anchor is a long-lived CA Certificate. The way it currently works is that certificate is passed into the `AuthenticateDocument` function call in `go-nitro-enclave-attestation-document`: https://github.com/veracruz-project/go-nitro-enclave-attestation-document/blob/0893a6c144113b9a29f3111101f64b4d6cf1c999/lib.go#L37

I did not want to embed a certificate, no matter how long-lived, in the logic of the module. Thus, it is being provided as part of the provisioning process.